great_expectations/datasource/fluent/interfaces.py

```diff
@@ -1317,16 +1319,22 @@ def compute_metrics(
                 )
                 results.append(MetricResultType(id=metric_id_for_batch, value=value))
             elif metric_id_for_batch in metrics_calculator_errors:
-                value = metrics_calculator_errors[metric_id_for_batch]
+                metrics_calculator_error = metrics_calculator_errors[metric_id_for_batch]
+                value = MetricErrorResultValue(
+                    exception_message=metrics_calculator_error["exception_info"].exception_message,
+                    exception_traceback=metrics_calculator_error[
+                        "exception_info"
+                    ].exception_traceback,
+                )
                 results.append(MetricErrorResult(id=metric_id_for_batch, value=value))
             else:
                 results.append(
-                    MetricInternalErrorResult(
```

Comment:
Now that this type is being added to the API, I decided to consolidate it with `MetricErrorResult`.